app/src/org/commcare/gis/DrawingBoundaryActivity.kt

```diff
@@ -196,11 +196,12 @@ class DrawingBoundaryActivity : BaseMapboxActivity(), LocationListener, MapboxMa
     }
 
     override fun onLocationChanged(location: Location) {
+        val prevLocation = previousLocation;
         if (location != null && location.accuracy <= locationMinAccuracy) {
-            val addLocation = previousLocation == null ||
```

Comment:
The previousLocation variable is a nullable Location but the distanceTo function won't accept null and generated a compile error. We actually null check previousLocation before entering the code block that calls distanceTo, but it's a class member so the compiler assumes it could be changed from somewhere else in the code. The fix was to simply retrieve the previousLocation into a local prevLocation variable, then null check and use that.